src/plugins/workspace/server/workspace_client.ts

```diff
@@ -53,9 +56,11 @@ export class WorkspaceClientWithSavedObject implements IWorkspaceDBImpl {
   ): ReturnType<IWorkspaceDBImpl['create']> {
     try {
       const { permissions, ...attributes } = payload;
+      const id = generateRandomId(WORKSPACE_ID_SIZE);
       const result = await this.getSavedObjectClientsFromRequestDetail(requestDetail).create<
         Omit<WorkspaceAttribute, 'id'>
       >(WORKSPACE_TYPE, attributes, {
+        id,
```

Comment:
I suppose we should do some retry logic here if the random ids collide with existing workspace id.

Reply:
base64 string with length 6 results in 68.7 billion ids, I think for workspace id, the chance of collision will be too small and doesn't worth to have a retry logic here. I'd let it to throw the error and let the user try again.

Reply:
@SuZhou-Joe 
> base64 string with length 6 results in 68.7 billion ids

I might be wrong on this assumption, the number of ids will be determined by `crypto.randomBytes(6)`, I'll double check

Reply:
Oh, since we do `slice(0, 6)` the base64url string in the end, so the maximum number of random string will always be 64^6 which is ~68.7 billion